### PR TITLE
Provide a friendlier error if data is stored as dtype 'category' (CTGAN, TVAE)

### LINKED ISSUE #1735: Provide a friendlier error if data is stored as dtype `'category'` (CTGAN, TVAE)
### Problem Description
Currently, SDV has only been tested extensively when the data is stored as `int64`, `float64`, `datetime` and `object`. Other dtypes are not guaranteed to work.

A frequent occurrence is having dtype `'category'`. Several users have run into issues when their data is stored as the `'category'` dtype. The error is not easy to understand.

```python
from sdv.single_table import CTGANSynthesizer

synthesizer = CTGANSynthesizer(metadata)
synthesizer.fit(data)
```
```
TypeError: Cannot interpret 'CategoricalDtype(categories=['BASIC', 'DELUXE', 'SUITE'], ordered=False)' as a data type
```

### Expected behavior
Until we are able to provide full support for the `'category'` dtype, we should provide a friendly error message to the user explaining that they have to cast this data as an object. This is only applicable to `CTGANSynthesizer` and `TVAESynthesizer` -- as only these synthesizers have problems with the dtype.

```python
synthesizer.fit(data)
```
```
InvalidDataError: Column 'gender' is stored as a 'category' type, which is not supported. Please cast this column to an 'object' to continue.
```

### Additional context
Root cause: [RDT #751](https://github.com/sdv-dev/RDT/issues/751)

### PR DESCRIPTION
**Pull request by frances-h**
Resolve #1735 
CU-86az0gxvn

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86az0gxvn SDV - Provide a friendlier error if data is stored as dtype `'category'` (CTGAN, TVAE) #1735](https://app.clickup.com/t/86az0gxvn)

File: sdv/errors.py [hunk 1]
```diff
@@ -56,5 +56,9 @@ def __str__(self):
         )
 
 
+class InvalidDataTypeError(Exception):
+    """Error to raise if data type is not valid."""
```
Comment (frances-h) [inline sdv/errors.py:60]: The `InvalidDataError` only accepts lists of errors and automatically formats it as: "The metadata has the following errors: ..." son instead I added a new error type.

**Review by fealho — APPROVED** [review]
LGTM!

**Review by R-Palazzo — APPROVED** [review]
LGTM!

**Pull request merged**